tests/helpers/kube.go

```diff
@@ -1577,65 +1713,252 @@ func VerifyEventsMatchPatterns(t *testing.T, ctx context.Context,
-					found = true
+	// Decode and create KWOK Stage
+	decoder := scheme.Codecs.UniversalDeserializer()
+	obj, _, err := decoder.Decode([]byte(stageYAML), nil, nil)
+	require.NoError(t, err, "failed to decode KWOK stage YAML")
 
-					if node.Status.Conditions[i].Status != status {
-						node.Status.Conditions[i].Status = status
-						node.Status.Conditions[i].LastTransitionTime = metav1.Now()
-						node.Status.Conditions[i].LastHeartbeatTime = metav1.Now()
-						modified = true
-					}
+	stage, ok := obj.(*kwokv1alpha1.Stage)
+	require.True(t, ok, "decoded object is not a KWOK Stage")
 
-					break
-				}
-			}
+	err = c.Resources().Create(ctx, stage)
+	require.NoError(t, err, "failed to create KWOK stage %s", stage.Name)
 
-			if !found {
-				now := metav1.Now()
-				node.Status.Conditions = append(node.Status.Conditions, v1.NodeCondition{
-					Type:               conditionType,
-					Status:             status,
-					LastTransitionTime: now,
-					LastHeartbeatTime:  now,
-					Reason:             "TestCondition",
-					Message:            "Set by test",
-				})
-				modified = true
-			}
+	t.Logf("✓ KWOK stage %s applied successfully", stage.Name)
+}
 
-			if !modified {
-				return nil
-			}
+// DeleteKwokStage removes a KWOK Stage by name (idempotent - won't fail if not found).
+func DeleteKwokStage(ctx context.Context, t *testing.T, c klient.Client, stageName string) {
+	t.Helper()
 
-			return client.Resources().UpdateStatus(ctx, node)
+	stage := &kwokv1alpha1.Stage{}
+	stage.SetName(stageName)
+
+	err := c.Resources().Delete(ctx, stage)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			t.Logf("KWOK stage %s already deleted", stageName)
+		} else {
+			t.Logf("Warning: failed to delete KWOK stage %s: %v", stageName, err)
+		}
+
+		return
+	}
+
+	t.Logf("✓ KWOK stage %s deleted successfully", stageName)
+}
+
+// DeleteAllLogCollectorJobs deletes all log-collector jobs in the nvsentinel namespace.
+// This is useful for cleaning up leftover jobs from previous tests.
+func DeleteAllLogCollectorJobs(ctx context.Context, t *testing.T, c klient.Client) {
+	t.Helper()
+
+	var jobList batchv1.JobList
+
+	err := c.Resources().List(ctx, &jobList, resources.WithLabelSelector("app=log-collector"))
+	if err != nil {
+		t.Logf("Warning: failed to list log-collector jobs: %v", err)
+		return
+	}
+
+	if len(jobList.Items) == 0 {
+		t.Logf("No log-collector jobs to clean up")
+		return
+	}
+
+	t.Logf("Deleting %d old log-collector job(s)", len(jobList.Items))
+
+	for _, job := range jobList.Items {
+		// Delete with PropagationPolicy=Background to also delete pods
+		deletePolicy := metav1.DeletePropagationBackground
+
+		err := c.Resources().Delete(ctx, &job, func(do *metav1.DeleteOptions) {
+			do.PropagationPolicy = &deletePolicy
 		})
+		if err != nil && !apierrors.IsNotFound(err) {
+			t.Logf("Warning: failed to delete job %s: %v", job.Name, err)
+		}
+	}
+
+	// Wait a bit for jobs to be fully deleted
+	time.Sleep(2 * time.Second)
```

Comment:
What if the job doesn't get deleted in 2 seconds? Can we instead use require.Eventually and check for jobs not found?